File: 1584-min-cost-to-connect-all-points/1584-min-cost-to-connect-all-points.java
class Solution {
    private int dis(int[] a, int[] b){
        return Math.abs(a[0]-b[0]) + Math.abs(a[1]-b[1]);
    }
    public int minCostConnectPoints(int[][] points) {
        int n = points.length;
        int[] minCost = new int[n];
        Arrays.fill(minCost, Integer.MAX_VALUE);
        minCost[0] = 0;
        int[] vis = new int[n];
        int curr = 0, ans = 0;
        while(curr >= 0){
            vis[curr] = 1;
            int minCurr=Integer.MAX_VALUE, nxtPoint = -1;
            for(int i=0; i<n; i++){
                if(curr==i || vis[i]==1) continue;
                minCost[i] = Math.min(dis(points[curr], points[i]), minCost[i]);
                if(minCurr > minCost[i]){
                    minCurr = minCost[i];
                    nxtPoint = i;
                }
            }
            ans += (minCurr == Integer.MAX_VALUE ? 0 : minCurr);
            curr = nxtPoint;
        }
        return ans;
    }
}

File: 1584-min-cost-to-connect-all-points/1584-min-cost-to-connect-all-points.cpp
class Solution {
public:
    int dis(vector<int> &a, vector<int>&b){
        return abs(a[0]-b[0])+abs(a[1]-b[1]);
    }
    int minCostConnectPoints(vector<vector<int>>& ps) {
        int n = ps.size();
        vector<int> vis(n, 0), minCost(n, INT_MAX);
        minCost[0] = 0;
        int curr = 0, ans = 0;
        while(curr >= 0){
            vis[curr] = 1;
            int currMin = INT_MAX, nxtPoint=-1;
            for(int i=0; i<n; i++){
                if(curr==i || vis[i]) continue;
                minCost[i] = min(dis(ps[curr], ps[i]), minCost[i]);
                if(currMin > minCost[i]){
                    currMin = minCost[i];
                    nxtPoint = i;
                }
            }
            ans += (currMin==INT_MAX ? 0 : currMin);
            curr = nxtPoint;
        }
        return ans;
    }
};